tests/drivers/charger/sbs_charger/testcase.yaml

```diff
@@ -0,0 +1,31 @@
+tests:
+  # section.subsection
+  drivers.sbs_charger.emulated:
+    tags: drivers charger
+    filter: >
+      dt_compat_enabled("sbs,sbs-charger") and
+      (CONFIG_QEMU_TARGET or CONFIG_BOARD_NATIVE_POSIX)
+    extra_args:
+      CONF_FILE="prj.conf;boards/emulated_board.conf"
+      DTC_OVERLAY_FILE="boards/emulated_board.overlay"
+    platform_exclude:
+      qemu_cortex_a53
+      qemu_cortex_a53_smp
+      qemu_kvm_arm64
+      xenvm
+      xenvm_gicv3
+      hifive_unmatched
+  drivers.sbs_charger.emulated_64_bit_i2c_addr:
+    tags: drivers charger
+    filter: >
+      dt_compat_enabled("sbs,sbs-charger") and
+      (CONFIG_QEMU_TARGET or CONFIG_BOARD_NATIVE_POSIX)
+    platform_allow:
+      qemu_cortex_a53
+      qemu_cortex_a53_smp
+      qemu_kvm_arm64
+      xenvm
+      xenvm_gicv3
+    extra_args:
+      CONF_FILE="prj.conf;boards/qemu_cortex_a53.conf"
+      DTC_OVERLAY_FILE="boards/qemu_cortex_a53.overlay"
```

Comment:
this looks quite complicated. why board specific overlays are being applied here?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/56666#discussion_r1170539146
I believe you asked this on April 18